narwhals/_compliant/dataframe.py

```diff
@@ -369,3 +395,55 @@ def _numpy_column_names(
         data: _2DArray, columns: Sequence[str] | None, /
     ) -> list[str]:
         return list(columns or (f"column_{x}" for x in range(data.shape[1])))
+
+    def _gather(self, item: SizedMultiIndexSelector[NativeSeriesT]) -> Self: ...
+    def _gather_slice(self, item: _SliceIndex | range) -> Self: ...
+    def _select_multi_index(
+        self, item: SizedMultiIndexSelector[NativeSeriesT]
+    ) -> Self: ...
+    def _select_multi_name(self, item: SizedMultiNameSelector[NativeSeriesT]) -> Self: ...
+    def _select_slice_index(self, item: _SliceIndex | range) -> Self: ...
+    def _select_slice_name(self, item: _SliceName) -> Self: ...
+    def __getitem__(
+        self,
+        item: tuple[
+            SingleIndexSelector | MultiIndexSelector[EagerSeriesT],
+            MultiIndexSelector[EagerSeriesT] | MultiColSelector[EagerSeriesT],
```

Comment:
```suggestion
            SingleColSelector | MultiColSelector[EagerSeriesT],
```
Same here?

Reply:
same response 😉  the `SingleColSelector` gets handled before we get here